### PR TITLE
Defining the GitHub webhook handler

### PR DESCRIPTION
**Pull request by susantoscott**
This PR contains the definition for a `GitHubWebhookHandler`. Given a [payload](https://docs.github.com/en/enterprise-server@3.3/rest/webhooks/repos#test-the-push-repository-webhook) from GitHub, which comes in the form of a `gh.PushEvent`, `GitHubWebhookHandler` receives this payload, processes it, and makes a call to `repoupdater.EnqueueRepoUpdate`. Along the way, it uses similar middleware to the [batch changes webhooks](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@githubWebhookHandler/-/blob/cmd/frontend/webhooks/github_webhooks.go), implementing the `Registerer` interface and using the same `ServeHTTP` method. This explains why the `github_webhook_handler.go` code is relatively short: as a `Registerer`, a lot of work including validating secrets, parsing config, and many more is already done in `cmd/frontend/webhooks/github_webhooks.go`.

This builds on the work of the background webhook worker from the [previous PR](https://github.com/sourcegraph/sourcegraph/pull/39833).

## Test plan
`go test -run TestGitHubWebhookHandle`

### CODE REVIEW

File: internal/repos/github_webhook_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	// And we're correctly getting a 'no such host' error
+	// as the server is not live
+	// This call to repo-updater is the last line of code
+	// to test before the end of the handler
+	if !(string(data) == `Post "http://repo-updater:3182/enqueue-repo-update": dial tcp: lookup repo-updater: no such host`+"\n") {
+		t.Fatal(errors.Newf(`expected body: [%s], got [%s]`,
+			`Post "http://repo-updater:3182/enqueue-repo-update": dial tcp: lookup repo-updater: no such host`,
+			string(data),
+		))
+	}
```
Comment (susantoscott) [inline internal/repos/github_webhook_handler_test.go:99]: Would you guys have a good recommendation on how I can test this? I realized this passes on my local machine but fails in BuildKite.

File: internal/repos/github_webhook_handler.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		return errors.Newf("expected GitHub.PushEvent, got %T", payload)
+	}
+
+	repoName := getNameFromEvent(event)
+	resp, err := repoupdater.DefaultClient.EnqueueRepoUpdate(ctx, repoName)
+	if err != nil {
+		return err
+	}
+
+	log.Scoped("GitHub handler", fmt.Sprintf("Successfully updated: %s", resp.Name))
```
Comment (varsanojidan) [inline internal/repos/github_webhook_handler.go:36]: >Github webhook handler

File: internal/repos/github_webhook_handler.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	return nil
+}
+
+func getNameFromEvent(event *gh.PushEvent) api.RepoName {
+	url := *event.Repo.URL
+	repoName := url[8:] // [ https:// ] accounts for 8 chars
```
Comment (varsanojidan) [inline internal/repos/github_webhook_handler.go:42]: You should add some sanity checks here that the length is more than 8 characters so we cant accidentally panic here.

File: internal/repos/github_webhook_handler_test.go (hunk 1 above)
Comment (varsanojidan) [reply internal/repos/github_webhook_handler_test.go:99]: You can use a vcr recorder to record the reply, or mock it out using HTTP Test Server if you don't have a good way to pass in the HTTP client for the VCR tests.

File: internal/repos/github_webhook_handler.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		return errors.Wrap(err, "handleGitHubWebhook: EnqueueRepoUpdate failed")
+	}
+
+	log.Scoped("GitHubWebhookhandler", fmt.Sprintf("Successfully updated: %s", resp.Name))
+	return nil
+}
+
+func getNameFromEvent(event *gh.PushEvent) (api.RepoName, error) {
+	url := *event.Repo.URL
+	if len(url) > 8 {
```
Comment (varsanojidan) [inline internal/repos/github_webhook_handler.go:46]: generally its more common practice to put the error condition in the if conditional, and make the final return be the success return.

**Review by varsanojidan — APPROVED** [review]
lgtm! left just one comment

**Pull request merged**